onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -292,22 +375,12 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
             params.Type = DFTType::Stockham;
             params.StockhamParams = {};
 
-            ComPtr<IUnknown> inputUnknown;
-            ComPtr<ID3D12Resource> inputResource;
-            inputTensor->GetDataInterface(inputUnknown.GetAddressOf());
-            inputUnknown.As(&inputResource);
-
-            ComPtr<IUnknown> outputUnknown;
-            ComPtr<ID3D12Resource> outputResource;
-            outputTensor->GetDataInterface(outputUnknown.GetAddressOf());
-            outputUnknown.As(&outputResource);
-
             // { before_dft_axis, axis, after_dft_axis, real_or_complex }
             std::array<uint32_t, 4> reshapedInputSize = { 1, 1, 1, inputDims.back() };
             std::array<uint32_t, 4> reshapedOutputSize = { 1, 1, 1, outputDims.back() };
 
             size_t reshapedIndex = 0;
-            for (int i = 0; i < inputDims.size() - 1; i++)
+            for (int i = 0; i < static_cast<int>(inputDims.size()) - 1; i++)
```

Comment:
>static_cast<int>(inputDims.size()) - 1 [](http://example.com/codeflow?start=32&length=38)

Propose `assert(!inputDims.empty())` above for underflow infinite loop sanity check.